tool/tsh/common/mcp_app.go

```diff
@@ -228,23 +228,53 @@ func newMCPServerWithDetails(app types.Application, accessChecker services.Acces
 	return a
 }
 
+type mcpListRBACPrinter struct {
+	showRBACFootnote bool
+}
+
+func (p *mcpListRBACPrinter) formatAllowedTools(mcpServer mcpServerWithDetails) string {
+	allowed := common.FormatAllowedEntities(mcpServer.Permissions.MCP.Tools.Allowed, mcpServer.Permissions.MCP.Tools.Denied)
+	if len(mcpServer.Permissions.MCP.Tools.Allowed) == 0 {
+		allowed += " [!]"
+		p.showRBACFootnote = true
+	}
+	return allowed
+}
+
+func (p *mcpListRBACPrinter) maybePrintHint(w io.Writer) error {
+	if !p.showRBACFootnote {
+		return nil
+	}
+	_, err := fmt.Fprintf(w, `[!] Warning: you do not have access to any MCP tools on the MCP server.
+Please contact your Teleport administrator to ensure your Teleport role has
+appropriate 'allow.mcp.tools' set. For details on MCP access RBAC, see:
+https://goteleport.com/docs/enroll-resources/mcp-access/rbac/
```

Comment:
this link does not exist yet but should be available next week when cloud goes to v18.

current link:
https://goteleport.com/docs/ver/18.x/enroll-resources/mcp-access/rbac/